datafusion/common/src/dfschema.rs

```diff
@@ -314,6 +314,42 @@ impl DFSchema {
         }
     }
 
+    /// Find if the field exists with the given name
+    pub fn has_column_with_unqualified_name(&self, name: &str) -> Result<bool> {
+        let matches = self.fields_with_unqualified_name(name);
+        match matches.len() {
+            0 => Ok(false),
+            1 => Ok(true),
+            _ => Err(DataFusionError::SchemaError(
+                SchemaError::AmbiguousReference {
+                    qualifier: None,
+                    name: name.to_string(),
```

Comment:
this `to_string()` is what kills performance (as it allocates a new string for every ambiguous error)